zilliqa/src/consensus.rs

```
@@ -177,9 +177,16 @@ pub struct Consensus {
     pub new_receipts: broadcast::Sender<(TransactionReceipt, usize)>,
     pub new_transactions: broadcast::Sender<VerifiedTransaction>,
     pub new_transaction_hashes: broadcast::Sender<Hash>,
+    /// Pruning interval i.e. how many blocks to keep in the database.
+    prune_interval: u64,
 }
 
 impl Consensus {
+    // gradual pruning interval
+    const PRUNE_INTERVAL: u64 = 1000;
+    // prune interval > 20 to ensure we don't prune forks
+    const MIN_PRUNE_INTERVAL: u64 = 20;
```

IMO we should validate the config to be greater than this (and fail to start if it isn't) instead of silently increasing the configured value.

---

Alright. I'll change the behavior to fail instead of clamping.